terragrunt/modules/wiz/main.tf

```diff
@@ -1,6 +1,205 @@
+          "profile:ListIntegrations",
+          "s3:GetBucketNotification",
+          "s3:GetMultiRegionAccessPointPolicy",
+          "s3:ListMultiRegionAccessPoints",
+          "ses:DescribeActiveReceiptRuleSet",
+          "ses:GetAccount",
+          "ses:GetConfigurationSet",
+          "ses:GetConfigurationSetEventDestinations",
+          "ses:GetDedicatedIps",
+          "ses:GetEmailIdentity",
+          "ses:ListConfigurationSets",
+          "ses:ListDedicatedIpPools",
+          "ses:ListReceiptFilters",
+          "voiceid:DescribeDomain",
+          "wafv2:GetLoggingConfiguration",
+          "wafv2:GetWebACLForResource",
+          "wisdom:GetAssistant",
+          "macie2:ListFindings",
+          "macie2:GetFindings",
+          "identitystore:List*",
+          "identitystore:Describe*",
+          "sso-directory:Describe*",
+          "sso-directory:ListMembersInGroup",
+          "cloudwatch:GetMetricStatistics"
+        ],
+        "Effect" : "Allow",
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "ec2:CopySnapshot",
+          "ec2:CreateSnapshot",
+          "kms:CreateKey",
+          "kms:DescribeKey",
+          "ec2:GetEbsEncryptionByDefault",
+          "ec2:DescribeSnapshots"
+        ],
+        "Effect" : "Allow",
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "ec2:CreateTags"
+        ],
+        "Effect" : "Allow",
+        "Resource" : "arn:aws:ec2:*::snapshot/*"
+      },
+      {
+        "Action" : "kms:CreateAlias",
+        "Effect" : "Allow",
+        "Resource" : [
+          "arn:aws:kms:*:*:alias/wizKey",
+          "arn:aws:kms:*:*:key/*"
+        ]
+      },
+      {
+        "Action" : [
+          "kms:CreateGrant",
+          "kms:ReEncryptFrom"
+        ],
+        "Condition" : {
+          "StringLike" : {
+            "kms:ViaService" : "ec2.*.amazonaws.com"
+          }
+        },
+        "Effect" : "Allow",
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "kms:GetKeyPolicy",
+          "kms:PutKeyPolicy"
+        ],
+        "Condition" : {
+          "StringEquals" : {
+            "aws:ResourceTag/wiz" : "auto-gen-cmk"
+          }
+        },
+        "Effect" : "Allow",
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "ec2:DeleteSnapshot",
+          "ec2:ModifySnapshotAttribute"
+        ],
+        "Condition" : {
+          "StringEquals" : {
+            "ec2:ResourceTag/wiz" : "auto-gen-snapshot"
+          }
+        },
+        "Effect" : "Allow",
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "s3:*",
+        ],
+        "Effect" : "Deny",
+        "Resource" : [
+          "arn:aws:s3:::*terraform*",
+          "arn:aws:s3:::*tfstate*",
+          "arn:aws:s3:::*tf?state*",
+          "arn:aws:s3:::*cloudtrail*",
+          "arn:aws:s3:::elasticbeanstalk-*",
+          "arn:aws:s3:::rust-release-keys",
```

Comment:
Can we allowlist instead or in addition to denylist, with an alarm/audit to make sure all buckets are listed?

Reply:
I'd argue with an allow list we would be losing benefit we get from Wiz s3 monitoring, which is getting insight into new buckets and making sure they were established correctly. I am more concerned about a bucket being stood up and used with unintentionally incorrect settings/permissions out-of-band than settings being changed on an existing bucket we know about (although that is also a concern). 

Are you suggesting we have a out-of-band job, maybe a validation lint here in `simpleinfra`, that confirms that the allowlist matches all buckets currently in existence mod explicit denials?

Reply:
Yes, I was under the impression that Wiz already is doing that validation - basically, I am 100% onboard with every bucket having an explicit decision, but I want that decision, at least while we still have buckets with sensitive contents. If we e.g. switch to using KMS encryption for terraform state and any other secrets in s3 rather than just s3 key encryption, then we wouldn't need this (can just not grant KMS key access so the objects aren't interesting).

Reply:
If I'm not missing anything, the only actions that Wiz is allowed to take on S3 are:

- `s3:GetBucketNotification`
- `s3:GetMultiRegionAccessPointPolicy`
- `s3:ListMultiRegionAccessPoints`

Those seem fine to have enabled for all buckets, even sensitive ones we create accidentally. Personally, I don't think it's worth the added complexity to maintain an allow list manually and create a script/GitHub Action that periodically checks the list against our AWS accounts.

Reply:
Now I'm confused, because those don't seem like enough to even get a list of buckets or check any of the policies I'd expect Wiz to want to check? Are we sure these permissions are a complete list?

Reply:
I totally missed that the AWS-managed [SecurityAudit](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/SecurityAudit.html) group also includes a bunch of permissions for S3.

```
"s3:GetAccelerateConfiguration",
"s3:GetAccessPoint",
"s3:GetAccessPointPolicy",
"s3:GetAccessPointPolicyStatus",
"s3:GetAccountPublicAccessBlock",
"s3:GetAnalyticsConfiguration",
"s3:GetBucket*",
"s3:GetEncryptionConfiguration",
"s3:GetInventoryConfiguration",
"s3:GetLifecycleConfiguration",
"s3:GetMetricsConfiguration",
"s3:GetObjectAcl",
"s3:GetObjectVersionAcl",
"s3:GetReplicationConfiguration",
"s3:ListAccessPoints",
"s3:ListAllMyBuckets",
```

Reply:
Okay, in that case it looks like Wiz isn't even requesting GetObject permissions -- maybe the right thing to do is for us to explicitly deny that permission and otherwise not need a deny list on these buckets?

Reply:
The Wiz-provided module had another resource that would request `GetObject` _if_ data scanning was enabled. But I removed that resource when copying over the policies, since we're not planning on enabled the feature anyways.

You mean changing the last block to something like this:

```json
{
  "Action": [
    "s3:GetObject"
  ],
  "Effect": "Deny",
  "Resource": [
    "*"
  ],
  "Sid": "WizAccessS3"
}
```

Reply:
Yes, exactly. I think that covers our bases and avoids needing bucket name matches (which could diverge accidentally).

Reply:
Done.